scipy/spatial/_voronoi.pyx

```diff
@@ -21,6 +21,7 @@ __all__ = ['sort_vertices_of_regions']
 # array-filling placeholder that can never occur
 DEF ARRAY_FILLER = -2
 
+@cython.boundscheck(False)
```

Comment:
It's not guaranteed internally that `len(remaining) >= shape`, so disabling the boundscheck is a bit dangerous.

If we know all callers follow that rule, then I'm fine with it.

Reply:
I'm certainly open to restoring bounds checking if someone can think of an input scenario (i.e., unit test) where this could happen.

The affected function only gets called twice with the same signature: 

```python
remaining_filter(remaining.shape[0], remaining, current_simplex)
```

I don't think you can easily cook up an example where `remaining` is shorter than its shape attribute, but I think I know what you mean -- would it be helpful to declare the function with a leading underscore to be explicit about its `_private` nature (only meant to be used in current module); i.e., attempt to approximate the `static` declaration in C?

Reply:
`cdef` functions aren't available from Python, so it's essentially a private function here.

In any case, it seems like the cleanest solution is to drop the first argument and turn off bounds-checking. There's no need to pass both `remaining` and `remaining.shape[0]`.